### PR TITLE
Support sampling_relevant attributes

### PR DESCRIPTION
**Pull request by lmolkova**
Related to https://github.com/open-telemetry/opentelemetry-specification/pull/1916

### CODE REVIEW

File: semantic-conventions/syntax.md [hunk 1]
```diff
@@ -7,17 +7,20 @@ Then, the semantic of each field is described.
 
 <!-- toc -->
 
-- [JSON Schema](#json-schema)
```
Comment (Oberon00) [inline semantic-conventions/syntax.md:10]: I think there is no need to regenerate the TOC.

**Comment by Oberon00** [comment]
It looks like you need to run the `black` automformatter again. Try `pip install -r dev-requirements.txt` and then `black .`

**Comment by Oberon00** [comment]
Please add a test to https://github.com/open-telemetry/build-tools/blob/main/semantic-conventions/src/tests/semconv/templating/test_markdown.py (plus accompanying input.md, expected.md and minimal yaml file with any name). Then this LGTM 👍

File: semantic-conventions/syntax.md (hunk 1 above)
Comment (lmolkova) [reply semantic-conventions/syntax.md]: vscode auto-regenerates it for me. Anyway, returned original one back

**Comment by lmolkova** [comment]
> Please add a test to https://github.com/open-telemetry/build-tools/blob/main/semantic-conventions/src/tests/semconv/templating/test_markdown.py (plus accompanying input.md, expected.md and minimal yaml file with any name). Then this LGTM 👍

I don't know when I'll get a chance to work on this. Simple spec update turned out to be way too complicated.

**Comment by lmolkova** [comment]
> > Please add a test to https://github.com/open-telemetry/build-tools/blob/main/semantic-conventions/src/tests/semconv/templating/test_markdown.py (plus accompanying input.md, expected.md and minimal yaml file with any name). Then this LGTM 👍
> 

Done!

File: semantic-conventions/syntax.md [hunk 2]
```diff
@@ -174,7 +174,6 @@ An attribute is defined by:
 - `sampling_relevant`, optional EXPERIMENTAL boolean,
```
Comment (mariojonke) [inline semantic-conventions/syntax.md:174]: is it still considered experimental?

**Review by mariojonke — COMMENTED** [review]
[changelog](https://github.com/open-telemetry/build-tools/blob/v0.6.0/semantic-conventions/CHANGELOG.md) and [version.py](https://github.com/open-telemetry/build-tools/blob/v0.6.0/semantic-conventions/src/opentelemetry/semconv/version.py) should probably be updated.

**Comment by Oberon00** [comment]
@mariojonke I can handle that in #67 or a follow-up.

**Review by Oberon00 — APPROVED** [review]
Thank you!

**Review by mariojonke — APPROVED** [review]

File: semantic-conventions/syntax.md (hunk 2 above)
Comment (lmolkova) [reply semantic-conventions/syntax.md:174]: I don't know, but they can stay experimental as far as I am concerned - all semantic conventions are experimental.

**Comment by lmolkova** [comment]
I updated the changelog and version. If it still looks good, can someone please merge it? Thanks!

**Comment by arminru** [comment]
@open-telemetry/specs-approvers This will affect the semantic conventions we have in the spec, please take a look.

**Comment by lmolkova** [comment]
> @open-telemetry/specs-approvers This will affect the semantic conventions we have in the spec, please take a look.

Current semantic conventions don't use `sampling_relevant` field, i.e. it does not change any of them.

Here's the change where I start using `sampling_relevant`, which is approved and waiting for this PR to fix the build.

**Comment by lmolkova** [comment]
@Oberon00 @arminru @mariojonke can someone please merge it?

**Comment by Oberon00** [comment]
Only the TC can merge PRS in this repo, same as the spec (so @arminru _can_ merge). IMHO we can merge this, because as you said the current spec does not use sampling_relevant.

**Comment by arminru** [comment]
Fair enough as the actual usage of it will still be reviewed in the spec.
Released as https://github.com/open-telemetry/build-tools/releases/tag/v0.7.0 so `otel/semconvgen:0.7.0` is ready for you to reference in your spec PR @lmolkova.

**Comment by lmolkova** [comment]
thank you all!

**Pull request merged**